bin/cli.ts

```diff
@@ -0,0 +1,143 @@
+#! /usr/bin/env bun
+
+import * as fs from 'fs';
+import { Command, Option } from '@commander-js/extra-typings';
+
+import { consola } from 'consola';
+import { parseProgram } from 'vm/program';
+import { CairoRunner, RunOptions } from 'runners/cairoRunner';
+
+consola.options = {
+  ...consola.options,
+  formatOptions: {
+    date: false,
+  },
+};
+
+const VERSION_CLI = '0.1.0';
+
+const program = new Command().name('cairo-vm-ts').version(VERSION_CLI);
+
+program
+  .command('run', { isDefault: true })
+  .description('Run a compiled Cairo program')
+  .argument('<program.json>', 'path to Cairo compilation artifacts')
+  .option('--no-relocate', 'do not relocate memory')
+  .addOption(
+    new Option(
+      '--offset <OFFSET>',
+      'start address of the relocated memory\nStarkWare verifier expects offset to be 1'
+    )
+      .default(0, '0')
+      .argParser(parseInt)
+  )
+  .option(
+    '--export-trace <TRACE_FILENAME>',
+    'export the trace, little-endian encoded'
+  )
+  .option(
+    '--export-memory <MEMORY_FILENAME>',
+    'export the relocated memory, little-endian encoded'
+  )
+  .option('--print-trace', 'print the trace')
+  .option('--print-memory', 'print the non-relocated memory')
+  .option('--print-relocated-memory', 'print the relocated memory')
+  .option('--print-output', 'print the output segment')
+  .action(async (path, options) => {
+    if (!path.match(/\.json$/)) throw new Error('Provided file is not a JSON');
+
+    const {
+      relocate,
+      offset,
+      exportMemory,
+      exportTrace,
+      printOutput,
+      printMemory,
+      printRelocatedMemory,
+      printTrace,
+    } = options;
+
+    if (
+      (!relocate && !!offset) ||
+      (!relocate && exportMemory) ||
+      (!relocate && printRelocatedMemory)
+    ) {
+      consola.error(
+        "option '--no-relocate' cannot be used with options '--offset <OFFSET>', '--export-memory <MEMORY_FILENAME>' or '--print-relocated-memory'"
+      );
+      process.exit(1);
+    }
+
+    consola.info(`Cairo VM TS ${VERSION_CLI} - Execution Mode`);
+
+    const program = parseProgram(fs.readFileSync(path, 'utf-8'));
+    const runner = new CairoRunner(program);
+    const config: RunOptions = {
+      relocate: relocate,
+      relocateOffset: offset,
```

Comment:
can't we keep the same naming ?

Reply:
Sure, but we'll need to either rename the flag `--offset` to `--relocate-offset` with is kinda long imo, or rename the `relocateOffset` property to `offset`. I think renaming `relocateOffset` is ok